st-app-cms/src/api/github-login/controllers/github-login.ts

```diff
@@ -0,0 +1,19 @@
+import LoginContext from "../interfaces/login-context";
+import BaseUser, { baseUserFields } from "../interfaces/user";
+import loginService from '../services/github-login';
+
+export default {
+  async create(ctx: LoginContext) {
+    const { githubId, email } = ctx.request.body;
+
+    if (!githubId || !email) {
+      return ctx.badRequest('Missing githubUsername or email');
+    }
+    try {
+      const user: BaseUser & {created: boolean} = await loginService.loginUser({ githubId, email });
+      return ctx.send(user, user.created ? 201 : 200);
```

Comment:
it will be more readable if you first calculate the status code, and then just pass it to send it like
```ts
const statusCode = user.created ? CREATED : OK;

return ctx.send(user, statusCode);
```